mllib/src/main/scala/org/apache/spark/mllib/evaluation/RegressionMetrics.scala

```diff
@@ -105,6 +112,14 @@ class RegressionMetrics @Since("1.2.0") (
    */
   @Since("1.2.0")
   def r2: Double = {
-    1 - SSerr / SStot
+    // In case of regression through the origin (biased case), the definition of R^2 is
+    // to be modified. Here is a review paper which explains why:
+    // J. G. Eisenhauer, Regression through the Origin. Teaching Statistics 25, 76–80 (2003)
+    // https://online.stat.psu.edu/~ajw13/stat501/SpecialTopics/Reg_thru_origin.pdf
```

Comment:
@iyounus Can you put this comment into the scaladoc instead? [This example](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/CentralMomentAgg.scala#L27) might be of use.